File: frontend/src/types/workflows.ts
export interface WorkflowStep {
  type: 'search_knowledge' | 'ai_summarize' | 'ai_qa' | 'slack_webhook' | 'http_webhook';
  label: string;
  config: Record<string, string | number | boolean>;
}

export type WorkflowStatus = 'active' | 'paused' | 'draft';
export type WorkflowTrigger = 'manual' | 'document_upload';

export interface WorkflowRun {
  id: string;
  workflowId: string;
  status: 'running' | 'success' | 'failed';
  triggerData: Record<string, unknown>;
  stepResults: StepResult[];
  error?: string;
  startedAt: string;
  completedAt?: string;
}

export interface StepResult {
  step: number;
  label: string;
  type: string;
  status: 'success' | 'failed';
  output?: string;
  outputs?: Record<string, unknown>;
  error?: string;
}

export interface Workflow {
  id: string;
  name: string;
  description?: string;
  triggerType: WorkflowTrigger;
  triggerConfig: Record<string, string>;
  steps: WorkflowStep[];
  status: WorkflowStatus;
  runCount: number;
  lastRunAt?: string;
  createdAt: string;
  updatedAt: string;
  recentRuns: WorkflowRun[];
}

File: backend/app/schemas/workflows.py
"""Workflow Pydantic schemas."""
from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel


class WorkflowStep(BaseModel):
    type: str  # search_knowledge | ai_summarize | ai_qa | slack_webhook | http_webhook
    label: str
    config: Dict[str, Any] = {}


class WorkflowCreate(BaseModel):
    name: str
    description: Optional[str] = None
    trigger_type: str = "manual"
    trigger_config: Dict[str, Any] = {}
    steps: List[WorkflowStep] = []
    status: str = "draft"


class WorkflowUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    trigger_type: Optional[str] = None
    trigger_config: Optional[Dict[str, Any]] = None
    steps: Optional[List[WorkflowStep]] = None
    status: Optional[str] = None


class WorkflowRunOut(BaseModel):
    id: UUID
    workflow_id: UUID
    status: str
    trigger_data: Dict[str, Any] = {}
    step_results: List[Any] = []
    error: Optional[str] = None
    started_at: datetime
    completed_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class WorkflowOut(BaseModel):
    id: UUID
    name: str
    description: Optional[str] = None
    trigger_type: str
    trigger_config: Dict[str, Any] = {}
    steps: List[Any] = []
    status: str
    run_count: int = 0
    last_run_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    recent_runs: List[WorkflowRunOut] = []

    model_config = {"from_attributes": True}
